plugins/game/lib/api/filters/KalturaUserScorePropertiesFilter.php

```diff
@@ -0,0 +1,327 @@
+<?php
+/**
+ * @package plugins.game
+ * @subpackage api.filters
+ * @abstract
+ * @relatedService UserScoreService
+ */
+class KalturaUserScorePropertiesFilter extends KalturaUserScorePropertiesBaseFilter
+{
+	/**
+	 * @var int
+	 */
+	public $gameObjectId;
+	
+	/**
+	 * @var KalturaGameObjectType
+	 */
+	public $gameObjectType;
+	
+	/**
+	 * @var string
+	 */
+	public $userIdEqual;
+	
+	/**
+	 * @var int
+	 */
+	public $placesAboveUser;
+	
+	/**
+	 * @var int
+	 */
+	public $placesBelowUser;
+	
+	
+	public function getMapBetweenObjects()
+	{
+		return array_merge(parent::getMapBetweenObjects(), self::$map_between_objects);
+	}
+	
+	protected function getCoreFilter()
+	{
+		return null;
+	}
+	
+	/**
+	 * Prepare the redis key to be called with
+	 * @return string
+	 * @throws KalturaAPIException
+	 */
+	protected function prepareGameObjectKey()
+	{
+		if (is_null($this->gameObjectId))
+		{
+			throw new KalturaAPIException(KalturaErrors::GAME_OBJECT_ID_REQUIRED);
+		}
+		if (!$this->gameObjectType)
+		{
+			throw new KalturaAPIException(KalturaErrors::GAME_OBJECT_TYPE_REQUIRED);
+		}
+		
+		$redisKey = kCurrentContext::getCurrentPartnerId();
+		$redisKey.= '_' . $this->gameObjectId;
+		KalturaLog::info("Accessing Redis game object: $redisKey");
+		return $redisKey;
+	}
+	
+	protected function getKUser($redisWrapper, $pUser)
+	{
+		$partner = kCurrentContext::getCurrentPartnerId();
+		$kUser = kuserPeer::getKuserByPartnerAndUid($partner, $pUser);
+		if (!$kUser)
+		{
+			throw new KalturaAPIException(KalturaErrors::USER_ID_NOT_FOUND, $pUser);
+		}
+		
+		return $kUser->getId();
+	}
+	
+	/**
+	 * Retrieves pUsers for all kUsers in the results array, and returns a map for these pUsers
+	 * @param $results
+	 * @return array
+	 * @throws PropelException
+	 */
+	protected function createMapKUserToPUser($results)
+	{
+		$kUsers = array_keys($results);
+		
+		$pUsers = kuserPeer::retrieveByPKs($kUsers);
+		if (!$pUsers)
+		{
+			KalturaLog::info('Failed to retrieve pUsers from DB');
+			return array();
+		}
+		
+		$mapKUserPUser = array();
+		foreach ($pUsers as $pUser)
+		{
+			$mapKUserPUser[$pUser->getId()] = $pUser->getPuserId();
+		}
+		
+		return $mapKUserPUser;
+	}
+	
+	/**
+	 * Adjust the results' ranks to be their real rank, instead of default Redis results starting from 0
+	 * The adjusted results are returned as an array of maps
+	 * @param $results
+	 * @param $startingRank
+	 * @param $mapKUserPUser
+	 * @return array
+	 */
+	protected function adjustRanksBasedOnStartingRank($results, $startingRank, $mapKUserPUser)
```

Comment:
since you also update the kuser-puser relation i would give more general name like formatUserScoreResults and explain in the comment before the function what you adjust